src/app/app-routing.module.ts

```diff
@@ -1,8 +1,15 @@
 import { NgModule } from '@angular/core';
 import { Routes, RouterModule } from '@angular/router';
+import { CartComponent, CheckoutComponent, ProductsComponent, ThankYouComponent } from './components';
 
 
-const routes: Routes = [];
+const routes: Routes = [
+  { path: 'cart', component: CartComponent },
+  { path: 'checkout', component: CheckoutComponent },
+  { path: 'products', component: ProductsComponent },
+  { path: 'thankyou', component: ThankYouComponent },
+  { path: '**', redirectTo: '' },
```

Comment:
```{ path: '**', redirectTo: '' }```
What's the behavior here?

Reply:
The ** path in the last route is a wildcard. The router will select this route if the requested URL doesn't match any paths for routes defined earlier in the configuration. I have it goi to the baseHref since there’s no PageNotFound component behavior.  So if you go to /foo this’ll just send you back to /.